src/index.ts

```diff
@@ -200,6 +210,78 @@ class MayanRouteBase<N extends Network> extends routes.AutomaticRoute<
       : getNativeContractAddress(tokenId.chain);
   }
 
+  async injectReferralInstructionsForSolana(
+    connection: Connection,
+    request: routes.RouteTransferRequest<N>,
+    instructions: TransactionInstruction[],
+    sender: PublicKey,
+    quote: MayanQuote,
+  ) {
+    const { fromChain } = request;
+    const referrerAddress = this.referrerAddress()?.solana;
+    const referralFee = this.getReferralParameters(request).referrerBps;
+
+    if (!referrerAddress || !referralFee || fromChain.network !== 'Mainnet') {
+      return instructions;
+    }
+
+    const referrer = new PublicKey(referrerAddress);
+    const finalInstructions: TransactionInstruction[] = [];
+    const mint = new PublicKey(quote.fromToken.mint);
+    const referrerAta = getAssociatedTokenAddressSync(mint, referrer, true);
```

Comment:
I think some of these web3.js functions dealing with tokens take in a tokenProgram parameter that can either be TOKEN_PROGRAM or TOKEN_2022_PROGRAM -- we need to get the program type of the `mint` and pass it to these functions or they may fail